fdbserver/TLogServer.actor.cpp

```diff
@@ -1830,7 +1838,7 @@ Future<Void> tLogPeekMessages(PromiseType replyPromise,
 	    .detail("Version", logData->version.get())
 	    .detail("RecoveredAt", logData->recoveredAt);
 	// Wait until we have something to return that the caller doesn't already have
-	if (logData->version.get() < reqBegin) {
+	if (!clusterRecoveryVersion.present() && logData->version.get() < reqBegin) {
```

Comment:
We should also block if cluster recovery version is present and "reqBegin" is above that version?

Reply:
`pullAsync` drives the cursor that does peeking. Its begin version is the RV. It will call the cursor as many times as needed to pull all the messages, each time with a larger version. We should not block `pullAsync` in this case.

Reply:
> Its begin version is the RV.

I am confused. Isn't it the case the logs peek versions till RV?

Reply:
And it is not possible to have any committed (and recoverable) versions above RV at the time of recovery?

Reply:
Sorry, I did not mean recovery begins at the RV, it is KCV + 1, of course.

No, I do not believe we can recover anything above the RV. The RV is determined by the cluster controller. No version above it should be recovered. If there is a misunderstanding here let's discuss offline.

Reply:
Per discussion, I will add an assert if reqBegin is greater than RV.